cdap-common/src/main/java/io/cdap/cdap/common/metadata/QueryParser.java

```diff
@@ -40,28 +43,121 @@ private QueryParser() {}
     List<QueryTerm> queryTerms = new ArrayList<>();
-    for (String term : Splitter.on(SPACE_SEPARATOR_PATTERN)
+    for (String inputTerm : Splitter.on(SPACE_SEPARATOR_PATTERN)
         .omitEmptyStrings().trimResults().split(query)) {
-      queryTerms.add(parseQueryTerm(term));
+      queryTerms.add(parseQueryTerm(inputTerm));
     }
     return queryTerms;
   }
 
+  /**
+   * Extracts the raw value of the input term, given that terms can follow a key:[comparison-operator]value syntax.
+   * This method removes any syntactic characters from the input string, including comparison and wildcard operators.
+   * As an example, extractTermValue("key:>=30") returns "30".
+   *
+   * Note that this method removes comparison operators from alphabetic strings as well, even though they do not qualify
+   * for numeric search.
+   * As an example, extractTermValue("+>=thirty") returns "thirty".
+   *
+   * @param term the search term, with all syntactic operators included
+   * @return the raw value of the search term, with all syntactic operators excluded
+   */
+  public static String extractTermValue(String term) {
+    term = lastSubTerm(term);
+    if (term.startsWith(">") || term.startsWith("<") || term.startsWith("=")) {
+      term = term.substring(1);
+    }
+    if (term.startsWith("=")) {
+      term = term.substring(1);
+    }
+    if (term.endsWith("*")) {
+      term = term.substring(0, term.length() - 1);
+    }
+
+    return term;
+  }
+
   private static QueryTerm parseQueryTerm(String term) {
-    if (term.charAt(0) == REQUIRED_OPERATOR && term.length() > 1) {
-      return new QueryTerm(term.substring(1), Qualifier.REQUIRED);
+    SearchType parsedSearchType = getSearchType(term);
+
+    Qualifier parsedQualifier = hasRequirementPrefix(term) || parsedSearchType == SearchType.NUMERIC
+        ? Qualifier.REQUIRED : Qualifier.OPTIONAL;
+
+    String parsedTerm = hasRequirementPrefix(term)
+        ? term.substring(1) : term;
+
+    Comparison parsedComparison = parsedSearchType == SearchType.STRING
+        ? Comparison.EQUALS : getComparison(term);
+
+    return new QueryTerm(parsedTerm, parsedQualifier, parsedSearchType, parsedComparison);
+  }
+
+  private static SearchType getSearchType(String term) {
+    // numeric searches must begin with an explicit comparison operator
+    String t = lastSubTerm(term);
+    if (!(t.startsWith(">")
+        || t.startsWith(">=")
+        || t.startsWith("<")
+        || t.startsWith("<=")
+        || t.startsWith("=="))) {
+      return SearchType.STRING;
+    }
+    try {
+      Double.parseDouble(extractTermValue(t));
+      return SearchType.NUMERIC;
+    } catch (NumberFormatException e) {
+      return SearchType.STRING;
+    }
+  }
+
+  private static boolean hasRequirementPrefix(String term) {
+    return (term.startsWith(REQUIRED_OPERATOR) && term.length() > 1);
+  }
+
+  private static String lastSubTerm(String term) {
+    if (term.startsWith(REQUIRED_OPERATOR) && term.length() > 1) {
```

Comment:
can replace this with `hasRequirementPrefix`

Reply:
Silly oversight. Addressed in the latest commit!